server/honeytrap.go

```diff
@@ -157,22 +157,39 @@ type ServiceMap struct {
 	Type string
 }
 
-func (hc *Honeytrap) findService(addr net.Addr, payload []byte) *ServiceMap {
+func (hc *Honeytrap) findService(conn net.Conn) *ServiceMap {
+	// Match on address first
+	for _, sm := range hc.matchers {
+		if !sm.Matcher(conn.LocalAddr()) {
+			continue
+		}
+		return sm
+	}
+
+	log.Debug("Couldn't match on addr, peeking connection %s => %s", conn.RemoteAddr(), conn.LocalAddr())
```

Comment:
Should we make peek configurable? Or only enable when we have services configured without port?

Reply:
I think the way it's set up now, it will never peek if all services have ports configured anyway. We could explicitly make it a configurable option (Under the listener config perhaps?) for clarity.